docs/source/guides/data.rst

```diff
@@ -0,0 +1,548 @@
+2. In the training loop you can pass multiple DataLoaders as a dict or list/tuple and Lightning
+   will automatically combine the batches from different DataLoaders.
+3. In the validation and test loop you have the option to return multiple DataLoaders,
+   which Lightning will call sequentially.
+
+
+Using LightningDataModule
+=========================
+You can set multiple DataLoaders in your :class:`~pytorch_lightning.core.datamodule.LightningDataModule`, and Lightning will handle the
+combination batch under-the-hood.
+
+TODO: add code snippet.
+
+Using LightningModule hooks
+===========================
+
+Concatenated DataSet
+--------------------
+For training with multiple Datasets you can create a :class:`~torch.utils.data.dataloader` class
+which wraps your multiple DataSets (this of course also works for testing and validation
+DataSets).
+
+(`reference <https://discuss.pytorch.org/t/train-simultaneously-on-two-DataSets/649/2>`_)
+
+.. testcode::
+
+    class ConcatDataSet(torch.utils.data.DataSet):
+        def __init__(self, *DataSets):
+            self.DataSets = DataSets
+
+        def __getitem__(self, i):
+            return tuple(d[i] for d in self.DataSets)
+
+        def __len__(self):
+            return min(len(d) for d in self.DataSets)
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+            concat_DataSet = ConcatDataSet(
+                DataSets.ImageFolder(traindir_A),
+                DataSets.ImageFolder(traindir_B)
+            )
+
+            loader = torch.utils.data.DataLoader(
+                concat_DataSet,
+                batch_size=args.batch_size,
+                shuffle=True,
+                num_workers=args.workers,
+                pin_memory=True
+            )
+            return loader
+
+        def val_dataloader(self):
+            # SAME
+            ...
+
+        def test_dataloader(self):
+            # SAME
+            ...
+
+Return multiple DataLoaders
+---------------------------
+You can set multiple DataLoaders in your :class:`~pytorch_lightning.core.lightning.LightningModule`, and Lightning will take care of batch combination.
+
+For more details please have a look at :paramref:`~pytorch_lightning.trainer.trainer.Trainer.Trainer.multiple_trainloader_mode`
+
+.. testcode::
+
+    class LitModel(LightningModule):
+
+        def train_dataloader(self):
+
+            loader_a = torch.utils.data.DataLoader(range(6), batch_size=4)
+            loader_b = torch.utils.data.DataLoader(range(15), batch_size=5)
+
+            # pass loaders as a dict. This will create batches like this:
+            # {'a': batch from loader_a, 'b': batch from loader_b}
+            loaders = {'a': loader_a,
+                       'b': loader_b}
+
+            # OR:
+            # pass loaders as sequence. This will create batches like this:
+            # [batch from loader_a, batch from loader_b]
+            loaders = [loader_a, loader_b]
+
+            return loaders
```

Comment:
```suggestion
        def train_dataloader(self):

            loader_a = torch.utils.data.DataLoader(range(6), batch_size=4)
            loader_b = torch.utils.data.DataLoader(range(15), batch_size=5)

            # pass loaders as a dict. This will create batches like this:
            # {'a': batch from loader_a, 'b': batch from loader_b}
            loaders = {'a': loader_a,
                       'b': loader_b}

            # OR:
            # pass loaders as sequence. This will create batches like this:
            # [batch from loader_a, batch from loader_b]
            loaders = [loader_a, loader_b]

            return loaders
            
        def training_step(self, batch, batch_idx):
             # access a list with a batch from each DataLoader
             batch_a, batch_b = batch
```